lib/ripper-tags.rb

```diff
@@ -35,17 +35,24 @@ def self.default_options
   class ForgivingOptionParser < OptionParser
     attr_accessor :ignore_unsupported_options
 
-    def parse(argv)
-      argv = argv.dup
+    def load_options_file(file)
+      @argv.unshift(*File.read(file).split(/\r?\n/).flat_map { |line|
+        line.match(/(=|\s)/)
+        ($1 == "" || $1 == "=") ? line : line.split(/\s+/, 2)
+      })
+    end
+
+    def parse!(argv, *)
       exceptions = []
-      remaining = []
+      @argv = argv
 
       while argv.size > 0
+        argv_orig = argv.dup
         begin
-          remaining = super(argv)
+          super(argv)
           break
         rescue OptionParser::InvalidOption => err
-          argv -= err.args
+          argv.replace(argv_orig - err.args)
```

Comment:
As-is, this won't work with unsupported options - because it replaces argv with argv_orig which still contains `--option FILE`. Thus throwing the code into an infinite loop.

Reply:
Also note that `argv_orig` never contained the newly inserted bad options, `remove` here actually does not do anything.

Reply:
See the hanging testcase [in my branch](https://github.com/k3rni/ripper-tags/commit/11b577862782cfd30be82d40c6c11027b240cce0#diff-b88772f3eb6a16d2b09c44e112553b1aR52), since I can't push to this one.

Reply:
Thanks for catching this! It seems like my approach isn't perfect either. I'll try a workaround.